### PR TITLE
Added alternative expression field in AWS CloudWatch scaler

### PR DESCRIPTION
**Pull request by dekelev**
Added alternative expression field in AWS CloudWatch scaler.
Here's an example ScaledObject config:
```
awsRegion: us-east-1
expression: >-
  SELECT MIN(MessageCount) FROM "AWS/AmazonMQ" WHERE Broker =
  'production' and Queue = 'worker'
metricCollectionTime: '60'
metricName: WorkerMessageCount
metricStatPeriod: '60'
minMetricValue: '0'
namespace: AWS/AmazonMQ
targetMetricValue: '100'
```

In addition, AWS CloudWatch scaler was changed to use int64 instead of float64 for metric type, since [HPA doesn't work with float numbers, integers only.](https://github.com/kedacore/keda/issues/2760#issuecomment-1075096499)

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* https://github.com/kedacore/keda-docs/pull/755
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

In addition to PR #2911 
Fixes https://github.com/kedacore/keda/issues/2998

### CODE REVIEW

**Comment by dekelev** [comment]
@JorTurFer Docs changes PR in v2.7:
https://github.com/kedacore/keda-docs/pull/755

**Review by JorTurFer — COMMENTED** [review]
LGTM!
Thanks for the contribution! ❤️ 
Only 2 things thing, could you update the changelog with the changes?
I have created [an issue](https://github.com/kedacore/keda/issues/2998) for tracking this, 
so the line that you have to add is something like this:
- **AWS CloudWatch:** Add support to use expressions([#2998](https://github.com/kedacore/keda/issues/2998))

(Under the ### Improvements section)

Please, rebase also the latest changes into your branch, there are some changes needed for running e2e tests

**Comment by dekelev** [comment]
> LGTM! Thanks for the contribution! ❤️ Only 2 things thing, could you update the changelog with the changes? I have created [an issue](https://github.com/kedacore/keda/issues/2998) for tracking this, so the line that you have to add is something like this:
> 
> * **AWS CloudWatch:** Add support to use expressions([#2998](https://github.com/kedacore/keda/issues/2998))
> 
> (Under the ### Improvements section)
> 
> **Please, rebase also the latest changes into your branch, there are some changes needed for running e2e tests**

Done

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 - **Prometheus Scaler:** Support for `X-Scope-OrgID` header ([#2667](https://github.com/kedacore/keda/issues/2667))
 - **RabbitMQ Scaler:** Include `vhost` for RabbitMQ when retrieving queue info with `useRegex` ([#2498](https://github.com/kedacore/keda/issues/2498))
 - **Selenium Grid Scaler:** Consider `maxSession` grid info when scaling. ([#2618](https://github.com/kedacore/keda/issues/2618))
+- **AWS CloudWatch:** Add support to use expressions([#2998](https://github.com/kedacore/keda/issues/2998))
```
Comment (JorTurFer) [inline CHANGELOG.md:74]: Sorry for being a pain :(
Could you sort this alphabetically? The convention we follow is: General at the top and below it, alphabetically sorted

Comment (dekelev) [reply CHANGELOG.md]: Done

File: pkg/scalers/aws_cloudwatch_scaler.go [hunk 2]
```diff
@@ -287,19 +280,27 @@ func (c *awsCloudwatchScaler) GetMetrics(ctx context.Context, metricName string,
...
+	if c.metadata.expression != "" {
+		metricNameSuffix = c.metadata.metricsName
+	} else {
+		metricNameSuffix = c.metadata.dimensionName[0]
+	}
+
 	externalMetric := &v2beta2.ExternalMetricSource{
 		Metric: v2beta2.MetricIdentifier{
-			Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString(fmt.Sprintf("aws-cloudwatch-%s", c.metadata.dimensionName[0]))),
+			Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString(fmt.Sprintf("aws-cloudwatch-%s", metricNameSuffix))),
```
Comment (zroubalik) [inline pkg/scalers/aws_cloudwatch_scaler.go:301]: I am not sure if we should do this, maybe we can just use this:
`Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString("aws-cloudwatch")),

Seems like that `c.metadata.metricsName` hasn't been used before, it is safe to drop it from here.
@JorTurFer WDYT?

**Review by zroubalik — COMMENTED** [review]
Generally looking good! Great job. Just a comment on metric name.

**Review by zroubalik — APPROVED** [review]
LGTM

**Review by JorTurFer — APPROVED** [review]
Sorry

**Review by JorTurFer — APPROVED** [review]
LGTM!
❤️ ❤️ ❤️ ❤️

**Pull request merged**